sympy/matrices/determinant.py

```diff
@@ -410,10 +409,30 @@ def _charpoly(M, x='lambda', simplify=_simplify):
             m = m * (x - simplify(i))
         return PurePoly(m, x)
 
-    berk_vector = _berkowitz_vector(M)
-    x = uniquely_named_symbol(x, berk_vector, modify=lambda s: '_' + s)
+    xdum = Dummy('t')
+    DOM = DomainMatrix.from_Matrix(M, radicals_limit=1)
+    domain = DOM.domain
+    if domain == EX and use_domain is not True:
+        use_domain = False
+
+    if use_domain is False:
+        berk_vector = _berkowitz_vector(M)
+        berk_vector = [simplify(a) for a in berk_vector]
+        dummy_poly = PurePoly(berk_vector, xdum)
+
+    else:
+        dom_vector = DOM.charpoly()
+        dummy_poly = PurePoly(dom_vector, xdum, domain=domain)
+
+        # XXX: calling simplify like this is a bad idea.
+        # The caller can simplify if they want
+        if domain.is_FractionField and not all(t.is_Symbol for t in getattr(domain, 'symbols', ())):
+            dummy_poly = PurePoly(simplify(dummy_poly.as_expr()), xdum)
```

Comment:
Can we get rid of this? Is it just the one test that needs to be changed? I don't think simplify should be used here.

Reply:
There is only one test that fails i.e. [test_eigen_slow](https://github.com/sympy/sympy/blob/master/sympy/matrices/tests/test_eigen.py#L142)
I have already added this test in the PR: [here](https://github.com/sympy/sympy/pull/21037/files#diff-fac8aefecf0b1888a82a1fe453f4def14de35db545295fa147d52a12758f887aR404-R405)

Reply:
Can the test just be changed? I don't think that `simplify` should be used as part of `charpoly` in general.

Reply:
Sure, will do.